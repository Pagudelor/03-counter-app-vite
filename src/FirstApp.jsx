import  PropTypes  from 'prop-types';

const newMessage= "paola";

const example1=()=>{
    return "example"
}
function example(){
    return "example2funcion de ejemplo2"
}
export const FirstApp =( {title, subTitle} )=>{
    return (
    <>
    <div data-testid="test-title">{title}</div>
    <p> { newMessage } </p>
    <h2>{subTitle}</h2>
    <p>{example()}</p>
    <h1>{example1()}</h1>
    <p>diojito</p>
    </>
    )
    }
FirstApp.propTypes = {
    title: PropTypes.string.isRequired
}
FirstApp.defaultProps = {
    subTitle:'hay hola soy el subtitulo',
    title: 'Hola soy el title'
}
