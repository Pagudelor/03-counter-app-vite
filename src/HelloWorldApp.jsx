export function App(){
    return <div>Hola Moto</div>;
}