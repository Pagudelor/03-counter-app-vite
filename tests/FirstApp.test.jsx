import  { render, screen }  from '@testing-library/react';
import{ FirstApp} from '../src/FirstApp'

describe('Pruebas en <FirstApp />', () => {
    const title = 'hola soy paola andrea';

    test('debe de hacer match con el snapshot', () => {
        const { container} = render(<FirstApp title= { title }/>)
        expect ( container ).toMatchSnapshot();
    });

    test('Debe mostrar el mensaje Hola Soy Paola Andrea', () => {
        render(<FirstApp title= { title }/>)
        expect (screen.getByText(title)).toBeTruthy();
    });
});