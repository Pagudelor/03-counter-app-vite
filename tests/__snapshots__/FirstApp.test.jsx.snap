// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    data-testid="test-title"
  >
    hola soy paola andrea
  </div>
  <p>
     
    paola
     
  </p>
  <h2>
    hay hola soy el subtitulo
  </h2>
  <p>
    example2funcion de ejemplo2
  </p>
  <h1>
    example
  </h1>
  <p>
    diojito
  </p>
</div>
`;
